crates/protocol/src/batch/span.rs

```diff
@@ -108,31 +108,36 @@ impl SpanBatch {
         l1_origins: &[BlockInfo],
         l2_safe_head: L2BlockInfo,
     ) -> Result<Vec<SingleBatch>, SpanBatchError> {
-        let mut single_batches = Vec::new();
         let mut origin_index = 0;
-        for batch in &self.batches {
-            if batch.timestamp <= l2_safe_head.block_info.timestamp {
-                continue;
-            }
-            let origin_epoch_hash = l1_origins[origin_index..l1_origins.len()]
-                .iter()
-                .enumerate()
-                .find(|(_, origin)| origin.number == batch.epoch_num)
-                .map(|(i, origin)| {
-                    origin_index = i;
-                    origin.hash
-                })
-                .ok_or(SpanBatchError::MissingL1Origin)?;
-            let single_batch = SingleBatch {
-                epoch_num: batch.epoch_num,
-                epoch_hash: origin_epoch_hash,
-                timestamp: batch.timestamp,
-                transactions: batch.transactions.clone(),
-                ..Default::default()
-            };
-            single_batches.push(single_batch);
-        }
-        Ok(single_batches)
```

Comment:
This change is not necessary. Could also move the `mut origin_index` into the try_fold but would be at expense of readability (singles vec turns into a tuple).

Happy to revert this change if you prefer the previous impl.

Reply:
I'd prefer to keep the previous format only because it's a bit more digestible and flatter nesting structure (less indentations that is).

Reply:
This likely won't need to change so shouldn't cause to much of a headache as long as we document well.

Reply:
Reverted, thanks!